### DISCUSSION TITLE
flutter firebase authentication 403 (Forbidden) error occured

### DISCUSSION THREAD

**Discussion by alexaung** [firebase/flutterfire#7924] [Q&A]
I am creating my flutter login screen with firebase authentication. Actually I can successfully login and create a user as well as google and facebook social sign in.

But today, I got the following error for sing in with email, google and fb. I did not change any thing yet. May I know where do I check?

    403. That’s an error.

    Your client does not have permission to get URL /identitytoolkit/v3/relyingparty/verifyPassword from this server. That’s all we know.

error from firebase

    `"[firebase_auth/unknown] com.google.firebase.FirebaseException: An internal error has 
    occurred. [ Json conversion failed! Failed to parse error for string [<!DOCTYPE html> 
    <html lang=en> <meta charset=utf-8> <meta name=viewport content="initial-scale=1, 
    minimum-scale=1, width=device-width"> <title>Error 403 (Forbidden)!!1</title> <style> 
    *{margin:0;padding:0}html,code{font:15px/22px arial,sans- 
    serif}html{background:#fff;color:#222;padding:15px}body{margin:7% auto 0;max- 
    width:390px;min-height:180px;padding:30px 0 15px}* > 
    body{background:url(//www.google.com/images/errors/robot.png) 100% 5px no- 
    repeat;padding-right:205px}p{margin:11px 0 22px;overflow:hidden}ins{color:#777;text- 
    decoration:none}a img{border:0}@media screen and (max-width:772px). 
    {body{background:none;margin-top:0;max-width:none;padding- 
    right:0}}#logo{background:url(//www.google.com/images/branding/googlelogo/1x/googlelogo_color_150x54dp.png) no- 
    repeat;margin-left:-5px}@media only screen and (min-resolution:192dpi). 
    {#logo{background:url(//www.google.com/images/branding/googlelogo/2x/googlelogo_color_150x54dp.png) no-repeat 0% 0%/100% 100%;-moz-border-image:url(//www.google.com/images/branding/googlelogo/2x/googlelogo_color_150x54dp.png) 0}}@media only screen and (-webkit-min-device-pixel-ratio:2){#logo{background:url(//www.google.com/images/branding/googlelogo/2x/googlelogo_color_150x54dp.png) no-repeat;-webkit-background-size:100% 100%}}#logo{display:inline-block;height:54px;width:150px} </style> <a href=//www.google.com/><span id=logo aria-label=Google></span></a> <p><b>403.</b> <ins>That’s an error.</ins> <p>Your client does not have permission to get URL <code>/identitytoolkit/v3/relyingparty/verifyAssertion</code> from this server. <ins>That’s all we know.</ins>] ]"
`

blog function

    `Future login(String email, String password) async {
     emit(const AuthLoginLoading());
     try {
       User? user = (await firebaseAuth.signInWithEmailAndPassword(
          email: email, password: password))
      .user;

      // login success state
      if (user != null) {
         emit(AuthLoginSuccess(user: user));
      }
    } on FirebaseAuthException catch (e) {
      emit(AuthLoginError(error: e.message!));
    }
    }`

flutter doctor -v

    `[✓] Flutter (Channel stable, 2.8.1, on macOS 12.1 21C52 darwin-x64, locale en-GB)
    • Flutter version 2.8.1 at /Users/aungmyooo/Development/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 77d935af4d (5 weeks ago), 2021-12-16 08:37:33 -0800
    • Engine revision 890a5fca2e
    • Dart version 2.15.1

    [✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/aungmyooo/Library/Android/sdk
    • Platform android-31, build-tools 31.0.0
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7281165)
    • All Android licenses accepted.

    [✓] Xcode - develop for iOS and macOS (Xcode 13.2.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • CocoaPods version 1.11.2

    [✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

    [✓] Android Studio (version 2020.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7281165)

    [✓] VS Code (version 1.63.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.32.0

    [✓] Connected device (3 available)
    • SM N975F (mobile)          • RF8M82Y3K9A                              • android-arm64  • Android 11 (API 30)
    • Aung Myo’s iPhone (mobile) • 06e02964e429eeebf29550b03ef955abe09891ef • ios            • iOS 15.2.1 19C63
    • Chrome (web)               • chrome                                   • web-javascript • Google Chrome 97.0.4692.71

    • No issues found!`

**Comment by sardsiders** [comment]
[firebase_auth/unkown]com.google.firebase.j an internal error has occurred.[json conversation failed!]failed to parse error for string [<!DOCTYPE html><html lang=en> <metacharset=utf-8> <meta name=viewportcontent="initial-scale=1 , minimum-scale=1,width=device-width"> <title>Error403(Forbidden)!!</title> <style> *{margin:0;padding:0}html,code{front}]

**Reply by hortigado** [reply]
Hi Can you solved this problem. I have the same problem.

**Comment by sardsiders** [comment]
i got the same error while im trying to sign in or register to my new created apk app, web works fine! dont know the problem yet . searched every where but no answers. i just know the problem is the firebase_auth . apk file some how not connecting to firebase_auth and 403 error  for that .why? i dont know!

**Comment by sardsiders** [comment]
let me lnow if you discover it

**Comment by andrewkimjoseph** [comment]
Same issue, any developments.

**Comment by andrewkimjoseph** [comment]
This error comes about for users who try to use your app in countries where the` ***.google.com domain` is restricted.

**Reply by jido2015** [reply]
It also happen when you are using VPN.

**Comment by sarojKumarPadhi** [comment]
does any one found any solution?

**Comment by andrewkimjoseph** [comment]
Hey, this issue comes about if you're trying to when you are sending requests to ***.google.com services (that includes Firebase) in countries where Google is restricted, like Iran for example. We can mark this as closed.
